### PR TITLE
Fixes problem hiding an empty map description

### PR DESCRIPTION
**Pull request by javierarce**
This PR fixes an issue with the map description. Currently if you enable the description in a map, and then edit the metadata to set it to an empty string, the description remains visible. This PR fixes that bug and add missing specs for the map dropdown. 

Original issue: https://github.com/CartoDB/cartodb/issues/3941

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5531/))

**Comment by javierarce** [comment]
Can you have a look, @xavijam?

File: lib/assets/test/spec/cartodb/table/map_options_dropdown.spec.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+describe("map_options_dropdown", function() {
+
+  beforeEach(function() {
+    var table = TestUtil.createTable('test');
+    var vis = TestUtil.createVis("jam");
```
Comment (xavijam) [inline lib/assets/test/spec/cartodb/table/map_options_dropdown.spec.js]: ja! jam, jamón...

File: lib/assets/javascripts/cartodb/table/map/map_options_dropdown.js [hunk 2]
```diff
@@ -138,20 +138,26 @@ cdb.admin.MapOptionsDropdown = cdb.ui.common.Dropdown.extend({
...
+          self.$(".description").addClass("disabled");
+        }
 
       } else if (_.contains(simpleOverlays, type)) {
 
         var display = overlay.get("display");
         self.mapOptions.set(type, display);
 
-        if (display) self.$el.find("." + type).removeClass("disabled");
+        if (display) self.$("." + type).removeClass("disabled");
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/map/map_options_dropdown.js]: No ifs inline please :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5532/))

**Comment by xavijam** [comment]
Apart from my comments, :es:.

File: lib/assets/javascripts/cartodb/table/map/map_options_dropdown.js (hunk 2 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb/table/map/map_options_dropdown.js]: Oh, it was old code, changed.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5534/))

**Pull request merged**